plugins/sqs/sqs.cpp

```diff
@@ -314,48 +339,93 @@ void SQSHPCCPlugin::SQSHPCC::setSQSConfiguration(const string& protocol, const s
 
 bool SQSHPCCPlugin::SQSHPCC::RegionExists(const string& region)
 {
+  
   if(region.empty())
     {
       return false;
     }
 
   const char *reg = region.c_str();
-
+  
   return (
           strieq(reg,"US_EAST_1") ||
-              strieq(reg,"US_WEST_1") ||
-              strieq(reg,"EU_WEST_1") ||
-              strieq(reg,"EU_CENTRAL_1") ||
-              strieq(reg,"AP_SOUTHEAST_1") ||
-               strieq(reg,"AP_SOUTHEAST_2")
-          );
+    	  strieq(reg,"US_WEST_1") ||
+          strieq(reg,"US_WEST_2") ||
+          strieq(reg,"US_EAST_2") ||
+          strieq(reg,"CA_CENTRAl_1") ||
+          strieq(reg,"EU_WEST_2") ||
+          strieq(reg,"AP_NORTHEAST_1") ||
+          strieq(reg,"AP_NORTHEAST_2") ||
+    	  strieq(reg,"EU_WEST_1") ||
+    	  strieq(reg,"EU_CENTRAL_1") ||
+    	  strieq(reg,"AP_SOUTHEAST_1") ||
+     	  strieq(reg,"AP_SOUTHEAST_2") ||
+  	  strieq(reg,"AP_SOUTH_1")
+	  ); 
 }
 
 
-const char *const SQSHPCCPlugin::SQSHPCC::getRegion(const string& region)
+const char *const SQSHPCCPlugin::SQSHPCC::getRegion(const string& region) 
 {
 
-  const char *reg = region.c_str();
-
+  const char *reg = region.c_str(); 
+    
   if(strieq(reg,"US_EAST_1"))  return Aws::Region::US_EAST_1;
   if(strieq(reg,"US_WEST_1")) return Aws::Region::US_WEST_1;
+  //if(strieq(reg,"US_EAST_2")) return Aws::Region::US_EAST_2;
+  if(strieq(reg,"US_WEST_2")) return Aws::Region::US_WEST_2;
   if(strieq(reg,"EU_WEST_1")) return Aws::Region::EU_WEST_1;
+  //if(strieq(reg,"EU_WEST_2")) return Aws::Region::EU_WEST_2;
   if(strieq(reg,"EU_CENTRAL_1")) return Aws::Region::EU_CENTRAL_1;
   if(strieq(reg,"AP_SOUTHEAST_1")) return Aws::Region::AP_SOUTHEAST_1;
   if(strieq(reg,"AP_SOUTHEAST_2"))  return Aws::Region::AP_SOUTHEAST_2;
+  if(strieq(reg,"AP_NORTHEAST_1")) return Aws::Region::AP_NORTHEAST_1;
+  if(strieq(reg,"AP_NORTHEAST_2")) return Aws::Region::AP_NORTHEAST_2;
+  if(strieq(reg,"AP_SOUTH_1")) return Aws::Region::AP_SOUTH_1;
+  //if(strieq(reg,"CA_CENTRAL_1")) return Aws::Region::CA_CENTRAL_1;
 
   throw string("Your region must be among these regions [ US_EAST_1, US_WEST_1, EU_WEST_1, EU_CENTRAL_1, AP_SOUTHEAST_1, AP_SOUTHEAST_2 ], please check your region");
 }
 
+
+
+
 /**
- *  These function expose the contract for ECL
+ *  These function expose the contract for ECL  
  *
  */
 
 namespace SQSHPCCPlugin
 {
 
-  ECL_SQS_API bool ECL_SQS_CALL publishMessage(ICodeContext * ctx,const char* region, const char* queueName, const char* message)
+  ECL_SQS_API bool ECL_SQS_CALL publishMessage(ICodeContext * ctx,const char* region, const char* queueName, const char* message,  bool useProxy, const char* proxyHost, __int32 proxyPort,const char* proxyUsername,const char* proxyPassword)
```

Comment:
This is almost a duplicate of publishOrderedMessage.  It would be sensible to common them up.